nemo/collections/llm/peft/lora.py

```diff
@@ -54,46 +66,117 @@ def __init__(
         lora_A_init_method='xavier',
         lora_dtype=None,
     ):
-        super(LinearAdapter, self).__init__()
         assert isinstance(orig_linear, nn.Linear)
+        super(LinearAdapter, self).__init__(
+            in_features=orig_linear.in_features,
+            out_features=orig_linear.out_features,
+            bias=orig_linear.bias is not None,
+            device=orig_linear.weight.device,
+            dtype=orig_linear.weight.dtype,
+        )
+        LinearAdapter._init_adapter(self, orig_linear)
 
-        self.orig_linear = orig_linear
-        self.dim = dim
-        self.scale = alpha / dim
+    @staticmethod
+    def _init_adapter(
+        obj,
+        orig_linear=None,
+        dim=8,
+        alpha=32,
+        dropout=0.1,
+        dropout_position='post',
+        lora_A_init_method='xavier',
+        lora_dtype=None,
+    ):
```

Comment:
I think it would be great if you can explain when is this function called with orig_linear = None, and when it's called with orig_linear not None. That will motivate the use of staticmethod with this `obj` argument